### PR TITLE
fix: Avoid reseting search when autoPlan is false

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**

Avoids reseting the search results in cases where `autoPlan` is false and `activeSearchId` might be null.

Also fixes a bug on mobile view where adjusting the search would revert to results without running the new search

To repro bug
- Use config with `autoPlan` set to false
- Plan trip
- Move location on map
- Go into mode selector and deselect a transit submode
- You should see trip search reverts back to trip results from step 2.

**PR Checklist:**
- [n/a] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [n/a] Are all languages supported (Internationalization/Localization)?
- [n/a] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/actions/ui.js [hunk 1]
```diff
@@ -279,7 +279,7 @@ export function handleBackButtonPress(e) {
     if (previousSearch) {
       // If back button pressed and active search has changed, set search to
       // previous search ID.
-      if (activeSearchId !== previousSearchId) {
+      if (activeSearchId && activeSearchId !== previousSearchId) {
```
Comment (binh-dam-ibigroup) [inline lib/actions/ui.js:282]: How does this have to do with `autoplan = false`?

Comment (amy-corson-ibigroup) [reply lib/actions/ui.js:282]: I think if the search changes, the activeSearchId changes, but if autoPlan is false, the activeSearchId still changes but is set to null, which is triggering this check.

File: percy/percy.test.js [hunk 2]
```diff
@@ -121,17 +121,9 @@ async function executeTest(page, isMobile, isCallTaker) {
     await page.click('label[title="Transit"]')
     await page.waitForTimeout(200)
 
-    // FIXME: Must click Edit again [mobile-specific]
```
Comment (binh-dam-ibigroup) [inline percy/percy.test.js:124]: Nice fix!

**Review by binh-dam-ibigroup — APPROVED** [review]
Nice job fixing the percy tests!

**Pull request merged**